modules/packages/Zarr.chpl

```diff
@@ -174,26 +194,37 @@ module Zarr {
     :throws Error: If the decompression fails
   */
   proc readChunk(param dimCount: int, chunkPath: string, chunkDomain: domain(dimCount), ref arraySlice: [] ?t) throws {
+    var s: stopwatch;
     const f: file;
     // if the file does not exist, the chunk is empty
+    if zarrProfiling then s.restart();
     try {
       f = open(chunkPath, ioMode.r);
     } catch {
-      arraySlice[arraySlice.domain] = 0;
+      arraySlice = 0;
       return;
     }
+    if zarrProfiling then times["Opening File, Read"].add(s.elapsed());
```

Comment:
consider adding helper functions for these to make the repeated patterns even simpler

Reply:
I will also consider this one down the line; it would probably require a static map of timers to make the patterns simpler which could get complicated considering that multiple threads and locales are all adding to the same duration sums.